### ISSUE TITLE
False positive on useless-super-delegation with *args

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#2270]
### Steps to reproduce

Run pylint on the following code:
```python
class Super:
    def __init__(self, *args):
        self.args = args

class Sub(Super):
    def __init__(self, a, b):
        super().__init__(a, b)
```

### Current behavior

The following message is issued:
```
6:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
```

### Expected behavior

I think no message should be issued, since the delegation is not useless: it forces the subclass to always take exactly 2 arguments, while the superclass can take any number of arguments.

This pattern occurs when defining a custom exception class, since the `Exception` class takes a variable number of arguments, but your custom exception type may have mandatory positional arguments.

### pylint --version output

```
pylint 2.0.0.dev2
astroid 2.0.0.dev4
Python 3.4.6 (default, Mar 22 2017, 12:26:13) [GCC]
```

**Comment by LewisGaul** [comment]
A similar but related case that gives a false positive warning:
```python
class Super:
    def __init__(self, **kwargs):
        self.kwargs = kwargs

class Sub(Super):
    def __init__(self, *, a, **kwargs):
        super().__init__(a, **kwargs)
```
Super class takes kwargs, subclass defines a specific arg and passes it through to the super class. Doesn't matter whether the subclass arg `a` has a default value or not.

**Comment by tuukkamustonen** [comment]
Also bumped into this. Still reproduces on:

```
pylint 2.9.6
astroid 2.6.6
Python 3.6.13 (default, May  7 2021, 10:50:49) 
[GCC 9.3.0]
```
